Kernel/Syscall.cpp

```diff
@@ -291,6 +291,15 @@ static u32 handle(RegisterDump& regs, u32 function, u32 arg1, u32 arg2, u32 arg3
         IO::out16(0x604, 0x2000);
         break;
     }
+    case Syscall::SC_reboot: {
+        dbgprintf("acquiring FS locks...\n");
```

Comment:
All of  this stuff should move into `Process::sys$reboot()` as well.
And `Process::sys$reboot()` should return `-EPERM` unless `is_superuser()`.